apps/metric_vnode/src/metric_io.erl

```diff
@@ -385,19 +390,29 @@ handle_cast({write, Bucket, Metric, Time, Value}, State) ->
 
 handle_cast({read, Bucket, Metric, Time, Count, ReqID, Sender},
             State = #state{node = N, partition = P}) ->
-    {D, State1} =
-        case get_set(Bucket, State) of
-            {ok, {{Resolution, MSet}, S2}} ->
-                {ok, Data} = mstore:get(MSet, Metric, Time, Count),
-                {{Resolution, Data}, S2};
-            _ ->
-                lager:warning("[IO] Unknown metric: ~p/~p", [Bucket, Metric]),
-                Resolution = get_resolution(Bucket),
-                {{Resolution, mmath_bin:empty(Count)}, State}
-        end,
+    {D, State1} = do_read(Bucket, Metric, Time, Count, State),
     riak_core_vnode:reply(Sender, {ok, ReqID, {P, N}, D}),
     {noreply, State1};
 
+handle_cast({read_rest, Bucket, Metric, Time, Count, Part, ReqID, Sender},
+            State = #state{node = N, partition = P}) ->
+    {Data, State1} =
+        case Part of
+            {Offset, PCount, Bin} when Offset =:= 0 ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time + PCount, Count - PCount, State),
```

Comment:
Line 402 is too long:                 {{Res, D}, S} = do_read(Bucket, Metric, Time + PCount, Count - PCount, State),.